pkg/cloud/services/network/routetables.go

```diff
@@ -302,8 +311,11 @@ func (s *Service) createRouteTableWithRoutes(routes []*ec2.Route, isPublic bool,
 			}
 			return true, nil
 		}, awserrors.RouteTableNotFound, awserrors.NATGatewayNotFound, awserrors.GatewayNotFound); err != nil {
-			// TODO(vincepri): cleanup the route table if this fails.
 			record.Warnf(s.scope.InfraCluster(), "FailedCreateRoute", "Failed to create route %s for RouteTable %q: %v", route.GoString(), *out.RouteTable.RouteTableId, err)
+			errDel := s.deleteRouteTable(out.RouteTable)
+			if errDel != nil {
+				s.scope.Trace("Failed to delete RouteTable %q: %v", *out.RouteTable.RouteTableId, errDel)
```

Comment:
Hey @vincepri  - is this a recommended approach. The idea here is not recording the route table delete failure as the `FailedCreateRoute` had been recorded. Is it make sense?

Reply:
+ @damdo